src/schemapack/_internals/spec/datapack.py

```diff
@@ -227,3 +227,68 @@ def validate_root_resource(self) -> Self:
                 {"root_resource": self.rootResource, "root_class": self.rootClass},
             )
         return self
+
+    @model_validator(mode="after")
+    def validate_relation_classes(self) -> Self:
+        """Checks if the `targetClass` of relations defined in a datapack exists
+        within datapack resources.
+        """
+        non_found_classes: dict[str, str] = {}  # target_class -> relation_name
+
+        for resources in self.resources.values():
+            for resource in resources.values():
+                if not resource.relations:
+                    continue
```

Comment:
Relations are always required, but can be an empty dict, right?
If that's the case, this check is unnecessary as this can just iterate over the empty dict